providers/cstor/cstor-block-device-pool/test.yml

```diff
@@ -15,178 +15,231 @@
-
-           - name: Initialize an empty list to store block device names
-             set_fact:
-               bd_names: []
-
-           - name: Store name of all block devices in the list
-             set_fact:
-               bd_names: "{{ bd_names + item.stdout_lines }}"
-             with_items:
-              - "{{ blockDevice.results }}"
-
-           - name: Adding discovered block device into SPC spec
-             lineinfile:
-               path: "./spc.yml"
-               insertafter: 'blockDeviceList:'
-               line: '    - {{ item }}'
-             with_items: "{{ bd_names }}"
-
-          when: pool_type == "raidz"
-
-        - block:
-
-           - name: Getting the Unclaimed block-device from each node when pool type is RAIDZ2
-             shell: kubectl get blockdevice -n openebs -l kubernetes.io/hostname={{ item }} -o jsonpath='{.items[?(@.status.claimState=="Unclaimed")].metadata.name}' | tr " " "\n" | grep -v sparse | head -n 6
-             register: blockDevice
-             with_items: "{{ nodes.stdout_lines }}"
-
-           - name: Initialize an empty list to store block device names
-             set_fact:
-               bd_names: []
-
-           - name: Store name of all block devices in the list
-             set_fact:
-               bd_names: "{{ bd_names + item.stdout_lines }}"
-             with_items:
-              - "{{ blockDevice.results }}"
-             
-           - name: Adding discovered block device into SPC spec
-             lineinfile:
-               path: "./spc.yml"
-               insertafter: 'blockDeviceList:'
-               line: '    - {{ item }}'
-             with_items: "{{ bd_names }}" 
           
-          when: pool_type == "raidz2"  
+        - block: 
+
+            - name: Getting the compute node name
+              shell: kubectl get nodes -l {{ node_label }} -o jsonpath='{.items[*].metadata.name}' | tr " " "\n"
+              register: nodes
+
+            - name: Replacing the pool type in SPC spec
+              replace:
+                path: ./spc.yml
+                regexp: "pool-type"
+                replace: "{{ pool_type }}"
+            
+            - block:
+
+              - name: Getting the Unclaimed block-device from each node when pool type is STRIPE
+                shell: kubectl get blockdevice -n openebs -l kubernetes.io/hostname={{ item }} -o jsonpath='{.items[?(@.status.claimState=="Unclaimed")].metadata.name}' | tr " " "\n" | grep -v sparse | head -n 1
+                register: blockDevice
+                with_items: "{{ nodes.stdout_lines }}"
+
+              - name: Initialize an empty list to store block device names
+                set_fact:
+                  bd_names: []
+
+              - name: Store name of all block devices in the list 
+                set_fact:
+                  bd_names: "{{ bd_names + item.stdout_lines }}"
+                with_items:
+                  - "{{ blockDevice.results }}"
+                  
```

Comment:
Good to print log in a case where we didn't find any blockDevices i.e if bd_names list is empty we need to print a message saying AutoPoolProvisioning helpful for debugging purpose.